client/src/partials/projects/project.js

```diff
@@ -1,117 +1,118 @@
+// TODO Handle HTTP exception errors (displayed contextually on form)
 angular.module('bhima.controllers')
 .controller('ProjectController', ProjectController);
 
 ProjectController.$inject = [
-  '$scope', '$window', 'validate', 'connect', 'messenger', 'appstate'
+   'ProjectService', 'EnterpriseService', 'SnisService', 'FormStateFactory'
 ];
 
-function ProjectController($scope, $window, validate, connect, messenger, appstate) {
-  var dependencies = {};
-  $scope.action = '';
-  $scope.timestamp = new Date();
-
-  dependencies.projects = {
-    query : {
-      tables : {
-        'project' : {
-          columns : ['id', 'name', 'abbr', 'enterprise_id']
-        }
-      }
-    }
-  };
-
-  dependencies.zs = {
-    query : {
-      tables : {
-        'mod_snis_zs' : {
-          columns : ['id', 'zone', 'territoire', 'province']
-        }
-      }
-    }
-  };
-
-  dependencies.enterprises = {
-    query : {
-      tables : {
-        'enterprise' : {
-          columns : ['id', 'name', 'abbr']
-        }
-      }
-    }
-  };
-
-  function handleErrors(err) {
-    messenger.danger('An Error occured : ' + err);
+function ProjectController(Projects, Enterprises, SnisService, StateFactory) {
+  var vm = this;
+
+  vm.enterprises = [];
+  vm.state = new StateFactory();
```

Comment:
In the future (not this PR), you probably shouldn't use `StateFactory()`.  It was an experiment that I created last Fall, and failed to a usable service.  The API is terrible and it causes more code confusion than clarity.

@sfount is investigating [ui-router](https://github.com/angular-ui/ui-router) to try and find a better way to manage state in the application.  If you can find a way to encapsulate state better, I'd be interested in seeing a proposal!